cfme/tests/openstack/cloud/test_volumes.py

```diff
@@ -0,0 +1,189 @@
+"""Tests for cloud provider's volumes"""
+
+import pytest
+from fauxfactory import gen_alphanumeric
+from cfme.cloud.instance import Instance
+from cfme.cloud.volume import (creation_form, device_input, get_volume_name,
+                               list_tbl, select_vm, select_volume, Volume)
+from cfme.fixtures import pytest_selenium
+from cfme.web_ui import InfoBlock, toolbar
+from cfme.web_ui.flash import assert_message_contain
+from cfme.web_ui.form_buttons import (attach_volume, detach_volume, FormButton,
+                                      submit_changes)
+from random import choice
+from utils import testgen, version
+from utils.appliance.implementations.ui import navigate_to
+from utils.wait import wait_for
+
+pytest_generate_tests = testgen.generate(testgen.provider_by_type,
+                                         ['openstack'], scope='module')
+
+
+PROD_VERSION = version.current_version().product_version()
+
+
+@pytest.mark.usefixtures("setup_provider_modscope")
+@pytest.mark.uncollectif(lambda: PROD_VERSION > '4.1')
+def test_create_volume(provider):
+    """Creates a volume for given cloud provider"""
+    navigate_to(Volume, 'All')
+    toolbar.select('Configuration', 'Add a new Cloud Volume')
+    vname = gen_alphanumeric()
+    vsize = 1
+    volume_data = dict(volume_name=vname,
+                       volume_size=vsize,
+                       cloud_tenant=provider.mgmt.list_tenant()[0])
+    creation_form.fill(volume_data)
+
+    pytest_selenium.click(FormButton('Add', ng_click='addClicked()'))
+    assert_message_contain('Create Cloud Volume')
+
+    wait_for(provider.is_refreshed, [None, 10], delay=5)
+    pytest_selenium.refresh()
+    params = {'Name': vname,
+              'Size': '{} GB'.format(vsize),
+              'Status': 'available'}
+    res = list_tbl.find_rows_by_cells(params)
+    assert res, 'Newly created volume doesn\'t appear in UI'
+
+
+@pytest.mark.uncollectif(lambda: PROD_VERSION > '4.1')
+def test_list_volumes_provider_details(provider):
+    """Verifies that all provider's volumes are present on details page"""
+    # Gather all provider's volumes name
+    volumes = []
+    for volume in provider.mgmt.list_volume():
+        vname = provider.mgmt.get_volume(volume).to_dict()['display_name']
+        volumes.append(vname)
+    provider.load_details()
+    err_msg = 'Details page shows wrong number of provider volumes'
+    clv_el = provider.summary.relationships.cloud_volumes
+    assert len(volumes) == clv_el.value, err_msg
```

Comment:
I think you can ignore the error messages here and just do the assertion. If you assert value comparison, list comparison, etc. pytest generates a suitable error message. Only in case the context of the assert might be unclear the message is good.